File: model/document/TransferFromShop.js
/**
 * Классы и типы для работы с актами передачи продукции.
 * @module module:transfer
 */
import ProductInfo from '../dictionary/ProductInfo'
import {TransferStatus} from '../types'

/**
 * @class module:transfer.TransferFromShop
 * @classdesc Акт передачи продукции из торгового зала на склад.
 * @param {string} uuid Уникальный идентификатор
 * @param {string} owner Кто подает документы
 * @param {?string} identity Идентификатор документа (клиентский)
 * @param {string} transferNumber Номер документа
 * @param {?Date} transferDate Дата составления
 * @param {?string} note Примечание
 * @param {module:transfer#TransferStatus} status Статус обработки документа
 * @param {?string} rejectComment Комментарий для отказа
 * @param {?string} replyId Уникальный идентификатор документа (присваивается УТМ); совпадает с идентификатором исходящего документа, который получили в ответе
 */
export default class TransferFromShop {
    constructor(uuid: string,
                owner: string,
                identity: (string | null),
                transferNumber: string,
                transferDate: (Date | null),
                note: (string | null),
                status: TransferStatus,
                rejectComment: (string | null),
                replyId: (string | null)) {
        this.uuid = uuid;
        this.owner = owner;
        this.identity = identity;
        this.transferNumber = transferNumber;
        this.transferDate = transferDate;
        this.note = note;
        this.status = status;
        this.rejectComment = rejectComment;
        this.replyId = replyId;
    }
}

/**
 * @class module:transfer.TransferFromShopPosition
 * @classdesc Позиция акта передачи продукции из торгового зала на склад.
 * @param {string} transferFromShopUuid Уникальный идентификатор документа передачи
 * @param {?string} identity Идентификатор позиции внутри акта
 * @param {string} productCode Регистрационный код продукции
 * @param {number} quantity Количество
 * @param {?string} informF2RegId Регистрационный номер раздела справки 2
 * @param {module:productInfo.ProductInfo} productInfo Информация о продукции
 */
export class TransferFromShopPosition {
    constructor(transferFromShopUuid: string,
                identity: (string | null),
                productCode: string,
                quantity: number,
                informF2RegId: (string | null),
                productInfo: ProductInfo) {
        this.transferFromShopUuid = transferFromShopUuid;
        this.identity = identity;
        this.productCode = productCode;
        this.quantity = quantity;
        this.informF2RegId = informF2RegId;
        this.productInfo = productInfo;
    }

}

File: android/src/main/java/com/evotor/egais/mapper/TransferFromShop.kt
package com.evotor.egais.mapper

import android.database.Cursor
import com.evotor.egais.converter.QuantityBigDecimalConverter
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap

internal fun createTransferFromShop(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("uuid", cursor.getString(cursor.getColumnIndexOrThrow("UUID")))
    result.putString("owner", cursor.getString(cursor.getColumnIndexOrThrow("OWNER")))
    result.putString("identity", cursor.getString(cursor.getColumnIndexOrThrow("IDENTITY")))
    result.putString("transferNumber", cursor.getString(cursor.getColumnIndexOrThrow("NUMBER")))
    result.putString("transferDate", cursor.getString(cursor.getColumnIndexOrThrow("TRANSFER_DATE")))
    result.putString("note", cursor.getString(cursor.getColumnIndexOrThrow("NOTE")))
    result.putString("status", cursor.getString(cursor.getColumnIndexOrThrow("STATUS")))
    result.putString("rejectComment", cursor.getString(cursor.getColumnIndexOrThrow("REJECT_COMMENT")))
    result.putString("replyId", cursor.getString(cursor.getColumnIndexOrThrow("REPLY_ID")))
    return result
}

internal fun createTransferFromShopPosition(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("transferFromShopUuid", cursor.getString(cursor.getColumnIndexOrThrow("TRANSFER_FROM_SHOP_ID")))
    result.putString("identity", cursor.getString(cursor.getColumnIndexOrThrow("IDENTITY")))
    result.putString("productCode", cursor.getString(cursor.getColumnIndexOrThrow("PRODUCT_CODE")))
    result.putDouble("quantity", QuantityBigDecimalConverter.toBigDecimal(cursor.getLong(cursor.getColumnIndexOrThrow("QUANTITY"))).toDouble())
    result.putString("informF2RegId", cursor.getString(cursor.getColumnIndexOrThrow("INFORM_F2_REG_ID")))
    result.putMap("productInfo", createProductInfo(cursor))
    return result
}